src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
 import { postControl, postRangeControl } from '../controllers/postControl';
 const router = express();
 
+/**
+ * @swagger
+ * /posts:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Get all posts
+ *     description: Retrieve a list of all posts with pagination, sorting, and filtering options.
+ *     parameters:
+ *       - in: query
+ *         name: perPage
+ *         schema:
+ *           type: integer
+ *           default: 15
+ *           maximum: 20
+ *         description: Number of posts to return per page (maximum is 20).
+ *       - in: query
+ *         name: page
+ *         schema:
+ *           type: integer
+ *           default: 1
+ *         description: The page number to retrieve.
+ *       - in: query
+ *         name: sort
+ *         schema:
+ *           type: string
+ *           default: createdAt
+ *         description: The field to sort the posts by.
+ *       - in: query
+ *         name: reverse
+ *         schema:
+ *           type: string
+ *           enum: [yes, no]
+ *           default: no
+ *         description: Whether to reverse the sort order (use 'yes' for descending order).
+ *     responses:
+ *       200:
+ *         description: A list of posts.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 type: object
+ *                 properties:
+ *                   _id:
+ *                     type: string
+ *                     description: The unique identifier of the post.
+ *                   title:
+ *                     type: string
+ *                     description: The title of the post.
+ *                   content:
+ *                     type: string
+ *                     description: The content of the post.
+ *                   createdAt:
+ *                     type: string
+ *                     format: date-time
+ *                     description: The creation timestamp of the post.
+ *                   likes:
+ *                     type: array
+ *                     items:
+ *                       type: string
+ *                     description: Array of user IDs who liked the post.
+ *                   creator_id:
+ *                     type: string
+ *                     description: The user ID of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message
+ */
 router.get('/', getAllPosts);
+
+/**
+ * @swagger
+ * /posts/getPostByID/{postID}:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Get a post by its ID
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"